{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true,
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "## TODO:\n",
    "- make column1 of the GSM DGE matrix files for the single cell\n",
    "data match column1 of the cell to cluster assignment file\n",
    "- transpose the GSM DGE matrix files so that the gene names are now column 1\n",
    "- join replicates (samp1+samp2, 1dpa1+1dpa2, 2dpa1+2dpa2, 4dpa1+4dpa2) on the gene name column\n",
    "- join all files together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Initialize samples and files\n",
    "Below are the sample names and the file names as arrays to be used later in the script."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "samples = ['samp1',\n",
    "           'samp2',\n",
    "           '1dpa1',\n",
    "           '1dpa2',\n",
    "           '2dpa1',\n",
    "           '2dpa2',\n",
    "           '4dpa1',\n",
    "           '4dpa2']\n",
    "files = ['GSM4095393_samp1_DGEmatrix.csv',\n",
    "         'GSM4095394_samp2_DGEmatrix.csv',\n",
    "         'GSM4095395_1dpa1_DGEmatrix.csv',\n",
    "         'GSM4095396_1dpa2_DGEmatrix.csv',\n",
    "         'GSM4095397_2dpa1_DGEmatrix.csv',\n",
    "         'GSM4095398_2dpa2_DGEmatrix.csv',\n",
    "         'GSM4095399_4dpa1_DGEmatrix.csv',\n",
    "         'GSM4095400_4dpa2_DGEmatrix.csv']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### `clean_matrix` function\n",
    "This funtion iterates over the rows of the dataframe and parses column 1 to get rid of\n",
    "the \"-1\" at the end, leaving the 16bp barcode. Then, it concatenates the sample identifier to the beginning.\n",
    "Next, outside the for loop, the dataframe is transposed. The transposition sets the index [0, 1, 2, ...]\n",
    "as column names, and the gene names as the indices. The next 3 lines set row 1 as the header\n",
    "so that the barcodes become the column names. Then, the last 2 lines before returning the final\n",
    "cleaned matrix resets the indices so that the gene names become their own column called \"genes\" and\n",
    "the indices are back to [0, 1, 2, ...]."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "def clean_matrix(exp, sample):\n",
    "    for index, row in exp.iterrows():\n",
    "        col1 = row['Barcodes']\n",
    "        col1 = col1[:16]\n",
    "        col1 = sample+col1\n",
    "        exp.at[index,'Barcodes'] = col1\n",
    "    exp = exp.transpose()\n",
    "    new_header = exp.iloc[0] #grab the first row for the header\n",
    "    exp = exp[1:] #take the data less the header row\n",
    "    exp.columns = new_header\n",
    "    exp.reset_index(inplace=True)\n",
    "    exp = exp.rename(columns = {'index':'genes'})\n",
    "    return exp"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### PreInjury samples\n",
    "The two preinjury files are processed and joined, then exported to a `csv.gz` file.\n",
    "The two matrices are joined in such a way that if a gene name is in one sample but\n",
    "not the other, the columns for that sample are filled with a `NaN` value."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "file = files[0]\n",
    "sample = samples[0]\n",
    "filepath = os.path.join('.', 'Hou_raw_matrices', file)\n",
    "samp1 = pd.read_csv(filepath)\n",
    "samp1 = clean_matrix(samp1, sample)\n",
    "\n",
    "file = files[1]\n",
    "sample = samples[1]\n",
    "filepath = os.path.join('.', 'Hou_raw_matrices', file)\n",
    "samp2 = pd.read_csv(filepath)\n",
    "samp2 = clean_matrix(samp2, sample)\n",
    "\n",
    "preinjury = pd.merge(samp1, samp2, left_on='genes', right_on='genes', how='outer')\n",
    "merged = os.path.join('.', 'Hou_expression_matrices', 'preinjury_merged.csv.gz')\n",
    "preinjury.to_csv(merged,\n",
    "           index=False,\n",
    "           compression=\"gzip\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "### 1dpa samples\n",
    "The two 1dpa files are processed and joined, then exported to a `csv.gz` file.\n",
    "The two matrices are joined in such a way that if a gene name is in one sample but\n",
    "not the other, the columns for that sample are filled with a `NaN` value."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "file = files[2]\n",
    "sample = samples[2]\n",
    "filepath = os.path.join('.', 'Hou_raw_matrices', file)\n",
    "samp1 = pd.read_csv(filepath)\n",
    "samp1 = clean_matrix(samp1, sample)\n",
    "\n",
    "file = files[3]\n",
    "sample = samples[3]\n",
    "filepath = os.path.join('.', 'Hou_raw_matrices', file)\n",
    "samp2 = pd.read_csv(filepath)\n",
    "samp2 = clean_matrix(samp2, sample)\n",
    "\n",
    "onedpa = pd.merge(samp1, samp2, left_on='genes', right_on='genes', how='outer')\n",
    "merged = os.path.join('.', 'Hou_expression_matrices', '_1dpa_merged.csv.gz')\n",
    "onedpa.to_csv(merged,\n",
    "           index=False,\n",
    "           compression=\"gzip\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "pycharm": {
     "name": "#%% md\n"
    }
   },
   "source": [
    "### 2dpa samples\n",
    "The two 2dpa files are processed and joined, then exported to a `csv.gz` file.\n",
    "The two matrices are joined in such a way that if a gene name is in one sample but\n",
    "not the other, the columns for that sample are filled with a `NaN` value."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "file = files[4]\n",
    "sample = samples[4]\n",
    "filepath = os.path.join('.', 'Hou_raw_matrices', file)\n",
    "samp1 = pd.read_csv(filepath)\n",
    "samp1 = clean_matrix(samp1, sample)\n",
    "\n",
    "file = files[5]\n",
    "sample = samples[5]\n",
    "filepath = os.path.join('.', 'Hou_raw_matrices', file)\n",
    "samp2 = pd.read_csv(filepath)\n",
    "samp2 = clean_matrix(samp2, sample)\n",
    "\n",
    "twodpa = pd.merge(samp1, samp2, left_on='genes', right_on='genes', how='outer')\n",
    "merged = os.path.join('.', 'Hou_expression_matrices', '_2dpa_merged.csv.gz')\n",
    "twodpa.to_csv(merged,\n",
    "           index=False,\n",
    "           compression=\"gzip\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 4dpa samples\n",
    "The two 4dpa files are processed and joined, then exported to a `csv.gz` file.\n",
    "The two matrices are joined in such a way that if a gene name is in one sample but\n",
    "not the other, the columns for that sample are filled with a `NaN` value."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "pycharm": {
     "name": "#%%\n"
    }
   },
   "outputs": [],
   "source": [
    "file = files[6]\n",
    "sample = samples[6]\n",
    "filepath = os.path.join('.', 'Hou_raw_matrices', file)\n",
    "samp1 = pd.read_csv(filepath)\n",
    "samp1 = clean_matrix(samp1, sample)\n",
    "\n",
    "file = files[7]\n",
    "sample = samples[7]\n",
    "filepath = os.path.join('.', 'Hou_raw_matrices', file)\n",
    "samp2 = pd.read_csv(filepath)\n",
    "samp2 = clean_matrix(samp2, sample)\n",
    "\n",
    "fourdpa = pd.merge(samp1, samp2, left_on='genes', right_on='genes', how='outer')\n",
    "merged = os.path.join('.', 'Hou_expression_matrices', '_4dpa_merged.csv.gz')\n",
    "fourdpa.to_csv(merged,\n",
    "           index=False,\n",
    "           compression=\"gzip\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Merge all samples\n",
    "\n",
    "First preinjury and 1dpa are merged. Then 2dpa and 4dpa are merged. Then 0+1 and 2+4 are merged making a final merged dataframe of all samples that is then exported to a `csv.gz` file. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "_0and1 = pd.merge(preinjury, onedpa, left_on='genes', right_on='genes', how='outer')\n",
    "_2and4 = pd.merge(twodpa, fourdpa, left_on='genes', right_on='genes', how='outer')\n",
    "full_merge = pd.merge(_0and1, _2and4, left_on='genes', right_on='genes', how='outer')\n",
    "merged = os.path.join('.', 'Hou_expression_matrices', 'full_merged.csv.gz')\n",
    "full_merge.to_csv(merged,\n",
    "           index=False,\n",
    "           compression=\"gzip\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Modifying `tximport` output\n",
    "We will be using the gene counts file from `tximport`, and it needs to match the formatting of the files above. The file is a tab delimited `.txt` file. The first code block reads in the file and gives the first column the same name as the first colummn in the expression matrix from above. In the second code block, the file is made into a `.csv.gz` to match the files created above. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "filepath = os.path.join('.', 'Danio_rerio.GRCz10.91.gene.counts.txt')\n",
    "gene_counts = pd.read_csv(filepath, sep='\\t')\n",
    "gene_counts = gene_counts.rename(columns = {'Unnamed: 0':'genes'})\n",
    "gene_counts.iloc[:3, :4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_count_file = os.path.join('.', 'Hartig_gene_counts.csv.gz')\n",
    "gene_counts.to_csv(gene_count_file,\n",
    "           index=False,\n",
    "           compression=\"gzip\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Making the Hou count file match the Hou design file\n",
    "In order for RNA Sieve to run, the count file has to match the design file to make the anndata object. It was discovered that the design file (`GSE137971_cells.csv.gz`) is missing 237 samples from the count file (`Hou_shared_gene_counts.csv.gz`). \n",
    "\n",
    "It should further be noted that the final gene count files that are being used in the RNA Sieve notebook were created at the command line with the following commands: \n",
    "\n",
    "```\n",
    "zcat Hartig_gene_counts.csv.gz Hou_expression_matrices/full_merged.csv.gz | cut -f 1 -d ',' |perl -ne \"s/gene/#gene/;print\" | sort | uniq -c | perl -ane \"if(@F[0]>1){print \\\"@F[1]\\n\\\"}\" | perl -ne \"s/#ge/ge/;print\" > shared_genes.txt\n",
    "\n",
    "zcat Hartig_gene_counts.csv.gz | perl -ne \"s/,/\\t/g;print\" | join shared_genes.txt - | perl -ne \"s/ /,/g;print\" > Hartig_shared_gene_counts.csv\n",
    "\n",
    "zcat Hou_expression_matrices/full_merged.csv.gz | perl -ne \"s/ge/#ge/;s/,/\\t/g;print\" | sort | perl -ne \"s/#ge/ge/;print\" | join shared_genes.txt - | perl -ne \"s/ /,/g;print\" > Hou_shared_gene_counts.csv\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>nCount_RNA</th>\n",
       "      <th>nFeature_RNA</th>\n",
       "      <th>Stage</th>\n",
       "      <th>S.Score</th>\n",
       "      <th>G2M.Score</th>\n",
       "      <th>major.cl</th>\n",
       "      <th>manualPhase</th>\n",
       "      <th>UMAP_1</th>\n",
       "      <th>UMAP_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1dpa1AAACCTGAGGGAAACA</td>\n",
       "      <td>6168</td>\n",
       "      <td>1292</td>\n",
       "      <td>1dpa</td>\n",
       "      <td>-0.045273</td>\n",
       "      <td>0.156326</td>\n",
       "      <td>Intermediate Epithelial</td>\n",
       "      <td>Unselected</td>\n",
       "      <td>-4.033391</td>\n",
       "      <td>5.024199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1dpa1AAACCTGCACACTGCG</td>\n",
       "      <td>6336</td>\n",
       "      <td>1388</td>\n",
       "      <td>1dpa</td>\n",
       "      <td>-0.034187</td>\n",
       "      <td>0.274646</td>\n",
       "      <td>Intermediate Epithelial</td>\n",
       "      <td>Unselected</td>\n",
       "      <td>-6.115250</td>\n",
       "      <td>1.632328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1dpa1AAACGGGGTATGAATG</td>\n",
       "      <td>9770</td>\n",
       "      <td>2034</td>\n",
       "      <td>1dpa</td>\n",
       "      <td>-0.050019</td>\n",
       "      <td>0.296948</td>\n",
       "      <td>Mesenchymal</td>\n",
       "      <td>Unselected</td>\n",
       "      <td>5.821854</td>\n",
       "      <td>-8.472561</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1dpa1AAAGATGAGCGCCTCA</td>\n",
       "      <td>8086</td>\n",
       "      <td>1782</td>\n",
       "      <td>1dpa</td>\n",
       "      <td>-0.041866</td>\n",
       "      <td>0.239046</td>\n",
       "      <td>Basal Epithelial</td>\n",
       "      <td>Unselected</td>\n",
       "      <td>7.921741</td>\n",
       "      <td>5.167769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1dpa1AAAGTAGAGGACACCA</td>\n",
       "      <td>5539</td>\n",
       "      <td>1382</td>\n",
       "      <td>1dpa</td>\n",
       "      <td>0.009987</td>\n",
       "      <td>0.036322</td>\n",
       "      <td>Intermediate Epithelial</td>\n",
       "      <td>Unselected</td>\n",
       "      <td>-6.163578</td>\n",
       "      <td>1.299275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18546</th>\n",
       "      <td>samp2TTTGGTTCATTCACTT</td>\n",
       "      <td>6871</td>\n",
       "      <td>1322</td>\n",
       "      <td>Preinjury</td>\n",
       "      <td>-0.004101</td>\n",
       "      <td>-0.008782</td>\n",
       "      <td>Intermediate Epithelial</td>\n",
       "      <td>G1</td>\n",
       "      <td>-1.582347</td>\n",
       "      <td>0.674932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18547</th>\n",
       "      <td>samp2TTTGTCACACCGGAAA</td>\n",
       "      <td>4421</td>\n",
       "      <td>762</td>\n",
       "      <td>Preinjury</td>\n",
       "      <td>-0.013394</td>\n",
       "      <td>-0.065488</td>\n",
       "      <td>Intermediate Epithelial</td>\n",
       "      <td>G1</td>\n",
       "      <td>-4.194117</td>\n",
       "      <td>-0.139754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18548</th>\n",
       "      <td>samp2TTTGTCACAGACAAAT</td>\n",
       "      <td>20003</td>\n",
       "      <td>2280</td>\n",
       "      <td>Preinjury</td>\n",
       "      <td>0.232262</td>\n",
       "      <td>0.082392</td>\n",
       "      <td>Mesenchymal</td>\n",
       "      <td>S</td>\n",
       "      <td>4.920971</td>\n",
       "      <td>-8.737699</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18549</th>\n",
       "      <td>samp2TTTGTCAGTTATGTGC</td>\n",
       "      <td>13829</td>\n",
       "      <td>1791</td>\n",
       "      <td>Preinjury</td>\n",
       "      <td>-0.019583</td>\n",
       "      <td>0.023622</td>\n",
       "      <td>Intermediate Epithelial</td>\n",
       "      <td>Unselected</td>\n",
       "      <td>-2.348434</td>\n",
       "      <td>5.227718</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18550</th>\n",
       "      <td>samp2TTTGTCAGTTGTGGCC</td>\n",
       "      <td>6858</td>\n",
       "      <td>1336</td>\n",
       "      <td>Preinjury</td>\n",
       "      <td>-0.002614</td>\n",
       "      <td>-0.056428</td>\n",
       "      <td>Superficial Epithelial</td>\n",
       "      <td>G1</td>\n",
       "      <td>-11.106893</td>\n",
       "      <td>-4.996488</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18551 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     samples  nCount_RNA  nFeature_RNA      Stage   S.Score  \\\n",
       "0      1dpa1AAACCTGAGGGAAACA        6168          1292       1dpa -0.045273   \n",
       "1      1dpa1AAACCTGCACACTGCG        6336          1388       1dpa -0.034187   \n",
       "2      1dpa1AAACGGGGTATGAATG        9770          2034       1dpa -0.050019   \n",
       "3      1dpa1AAAGATGAGCGCCTCA        8086          1782       1dpa -0.041866   \n",
       "4      1dpa1AAAGTAGAGGACACCA        5539          1382       1dpa  0.009987   \n",
       "...                      ...         ...           ...        ...       ...   \n",
       "18546  samp2TTTGGTTCATTCACTT        6871          1322  Preinjury -0.004101   \n",
       "18547  samp2TTTGTCACACCGGAAA        4421           762  Preinjury -0.013394   \n",
       "18548  samp2TTTGTCACAGACAAAT       20003          2280  Preinjury  0.232262   \n",
       "18549  samp2TTTGTCAGTTATGTGC       13829          1791  Preinjury -0.019583   \n",
       "18550  samp2TTTGTCAGTTGTGGCC        6858          1336  Preinjury -0.002614   \n",
       "\n",
       "       G2M.Score                 major.cl manualPhase     UMAP_1    UMAP_2  \n",
       "0       0.156326  Intermediate Epithelial  Unselected  -4.033391  5.024199  \n",
       "1       0.274646  Intermediate Epithelial  Unselected  -6.115250  1.632328  \n",
       "2       0.296948              Mesenchymal  Unselected   5.821854 -8.472561  \n",
       "3       0.239046         Basal Epithelial  Unselected   7.921741  5.167769  \n",
       "4       0.036322  Intermediate Epithelial  Unselected  -6.163578  1.299275  \n",
       "...          ...                      ...         ...        ...       ...  \n",
       "18546  -0.008782  Intermediate Epithelial          G1  -1.582347  0.674932  \n",
       "18547  -0.065488  Intermediate Epithelial          G1  -4.194117 -0.139754  \n",
       "18548   0.082392              Mesenchymal           S   4.920971 -8.737699  \n",
       "18549   0.023622  Intermediate Epithelial  Unselected  -2.348434  5.227718  \n",
       "18550  -0.056428   Superficial Epithelial          G1 -11.106893 -4.996488  \n",
       "\n",
       "[18551 rows x 10 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# adding a label to the first column & sorting to see if it will fix the anndata mismatching issue\n",
    "design = pd.read_csv(\"GSE137971_cells.csv.gz\", compression=\"gzip\")\n",
    "design = design.rename(columns = {'Unnamed: 0':'samples'})\n",
    "design = design.sort_values(design.columns[0])\n",
    "design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "design_file = os.path.join('.', 'sorted_labeled_GSE137971_cells.csv.gz')\n",
    "design.to_csv(design_file,\n",
    "           index=False,\n",
    "           compression=\"gzip\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# grab the first column of the design file to use as reference \n",
    "design = design.iloc[:,:1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read in the count file\n",
    "counts = pd.read_csv(\"Hou_shared_gene_counts.csv.gz\", compression=\"gzip\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# transpose the count file and set the header line and the index appropriately \n",
    "counts = counts.transpose()\n",
    "new_header = counts.iloc[0] #grab the first row for the header\n",
    "counts = counts[1:] #take the data less the header row\n",
    "counts.columns = new_header\n",
    "counts.reset_index(inplace=True)\n",
    "counts = counts.rename(columns = {'index':'samples'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>samples</th>\n",
       "      <th>ENSDARG00000000001</th>\n",
       "      <th>ENSDARG00000000002</th>\n",
       "      <th>ENSDARG00000000018</th>\n",
       "      <th>ENSDARG00000000019</th>\n",
       "      <th>ENSDARG00000000068</th>\n",
       "      <th>ENSDARG00000000069</th>\n",
       "      <th>ENSDARG00000000086</th>\n",
       "      <th>ENSDARG00000000103</th>\n",
       "      <th>ENSDARG00000000142</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSDARG00000109171</th>\n",
       "      <th>ENSDARG00000109173</th>\n",
       "      <th>ENSDARG00000109175</th>\n",
       "      <th>ENSDARG00000109179</th>\n",
       "      <th>ENSDARG00000109181</th>\n",
       "      <th>ENSDARG00000109183</th>\n",
       "      <th>ENSDARG00000109184</th>\n",
       "      <th>ENSDARG00000109185</th>\n",
       "      <th>ENSDARG00000109187</th>\n",
       "      <th>ENSDARG00000109188</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4253</th>\n",
       "      <td>1dpa1AAACCTGAGGGAAACA</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4254</th>\n",
       "      <td>1dpa1AAACCTGCACACTGCG</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4255</th>\n",
       "      <td>1dpa1AAACGGGGTATGAATG</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4256</th>\n",
       "      <td>1dpa1AAAGATGAGCGCCTCA</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4257</th>\n",
       "      <td>1dpa1AAAGTAGAGGACACCA</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4248</th>\n",
       "      <td>samp2TTTGGTTCATTCACTT</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4249</th>\n",
       "      <td>samp2TTTGTCACACCGGAAA</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4250</th>\n",
       "      <td>samp2TTTGTCACAGACAAAT</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4251</th>\n",
       "      <td>samp2TTTGTCAGTTATGTGC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4252</th>\n",
       "      <td>samp2TTTGTCAGTTGTGGCC</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18551 rows × 24325 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    samples ENSDARG00000000001 ENSDARG00000000002  \\\n",
       "4253  1dpa1AAACCTGAGGGAAACA                  0                  0   \n",
       "4254  1dpa1AAACCTGCACACTGCG                  0                  0   \n",
       "4255  1dpa1AAACGGGGTATGAATG                  0                  1   \n",
       "4256  1dpa1AAAGATGAGCGCCTCA                  0                  0   \n",
       "4257  1dpa1AAAGTAGAGGACACCA                  0                  0   \n",
       "...                     ...                ...                ...   \n",
       "4248  samp2TTTGGTTCATTCACTT                  0                  0   \n",
       "4249  samp2TTTGTCACACCGGAAA                  0                  0   \n",
       "4250  samp2TTTGTCACAGACAAAT                  0                  0   \n",
       "4251  samp2TTTGTCAGTTATGTGC                  0                  0   \n",
       "4252  samp2TTTGTCAGTTGTGGCC                  0                  0   \n",
       "\n",
       "     ENSDARG00000000018 ENSDARG00000000019 ENSDARG00000000068  \\\n",
       "4253                  0                  0                  0   \n",
       "4254                  0                  0                  0   \n",
       "4255                  0                  0                  0   \n",
       "4256                  0                  0                  0   \n",
       "4257                  0                  0                  0   \n",
       "...                 ...                ...                ...   \n",
       "4248                  0                  0                  0   \n",
       "4249                  0                  0                  0   \n",
       "4250                  0                  1                  0   \n",
       "4251                  0                  1                  0   \n",
       "4252                  0                  0                  0   \n",
       "\n",
       "     ENSDARG00000000069 ENSDARG00000000086 ENSDARG00000000103  \\\n",
       "4253                  0                  0                  0   \n",
       "4254                  0                  0                  0   \n",
       "4255                  0                  0                  0   \n",
       "4256                  5                  0                  0   \n",
       "4257                  0                  0                  0   \n",
       "...                 ...                ...                ...   \n",
       "4248                  0                  0                  0   \n",
       "4249                  0                  0                  0   \n",
       "4250                  0                  0                  0   \n",
       "4251                  0                  0                  0   \n",
       "4252                  2                  0                  0   \n",
       "\n",
       "     ENSDARG00000000142  ... ENSDARG00000109171 ENSDARG00000109173  \\\n",
       "4253                  0  ...                  0                  0   \n",
       "4254                  0  ...                  0                  0   \n",
       "4255                  0  ...                  0                  0   \n",
       "4256                  0  ...                  0                  0   \n",
       "4257                  0  ...                  0                  0   \n",
       "...                 ...  ...                ...                ...   \n",
       "4248                  0  ...                  0                  0   \n",
       "4249                  0  ...                  0                  0   \n",
       "4250                  0  ...                  0                  0   \n",
       "4251                  0  ...                  0                  0   \n",
       "4252                  0  ...                  0                  0   \n",
       "\n",
       "     ENSDARG00000109175 ENSDARG00000109179 ENSDARG00000109181  \\\n",
       "4253                NaN                NaN                  0   \n",
       "4254                NaN                NaN                  0   \n",
       "4255                NaN                NaN                  0   \n",
       "4256                NaN                NaN                  0   \n",
       "4257                NaN                NaN                  0   \n",
       "...                 ...                ...                ...   \n",
       "4248                NaN                NaN                  0   \n",
       "4249                NaN                NaN                  0   \n",
       "4250                NaN                NaN                  0   \n",
       "4251                NaN                NaN                  0   \n",
       "4252                NaN                NaN                  0   \n",
       "\n",
       "     ENSDARG00000109183 ENSDARG00000109184 ENSDARG00000109185  \\\n",
       "4253                NaN                  0                  0   \n",
       "4254                NaN                  0                  0   \n",
       "4255                NaN                  0                  0   \n",
       "4256                NaN                  0                  0   \n",
       "4257                NaN                  0                  0   \n",
       "...                 ...                ...                ...   \n",
       "4248                NaN                  0                  0   \n",
       "4249                NaN                  0                  0   \n",
       "4250                NaN                  0                  0   \n",
       "4251                NaN                  0                  0   \n",
       "4252                NaN                  0                  0   \n",
       "\n",
       "     ENSDARG00000109187 ENSDARG00000109188  \n",
       "4253                  0                  0  \n",
       "4254                  0                  0  \n",
       "4255                  0                  0  \n",
       "4256                  0                  0  \n",
       "4257                  0                  0  \n",
       "...                 ...                ...  \n",
       "4248                  0                  0  \n",
       "4249                  0                  0  \n",
       "4250                  0                  0  \n",
       "4251                  0                  0  \n",
       "4252                  0                  0  \n",
       "\n",
       "[18551 rows x 24325 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# performing a merge on just the sample names from the design file and the full count file \n",
    "# will result in a dataframe that only contains the count daata from the samples that exist in the design file\n",
    "joined = pd.merge(counts, design, left_on='samples', right_on='samples')\n",
    "# sort the joined to match the design \n",
    "joined = joined.sort_values(joined.columns[0])\n",
    "joined = joined.iloc[:, :-9]\n",
    "joined"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# export the count file that now matches the design file without transposing\n",
    "joined_file = os.path.join('.', 'sorted_long_matched_Hou_shared_gene_counts.csv.gz')\n",
    "joined.to_csv(joined_file,\n",
    "           index=False,\n",
    "           compression=\"gzip\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>samples</th>\n",
       "      <th>genes</th>\n",
       "      <th>1dpa1AAACCTGAGGGAAACA</th>\n",
       "      <th>1dpa1AAACCTGCACACTGCG</th>\n",
       "      <th>1dpa1AAACGGGGTATGAATG</th>\n",
       "      <th>1dpa1AAAGATGAGCGCCTCA</th>\n",
       "      <th>1dpa1AAAGTAGAGGACACCA</th>\n",
       "      <th>1dpa1AAAGTAGGTGTCGCTG</th>\n",
       "      <th>1dpa1AACGTTGGTGACGCCT</th>\n",
       "      <th>1dpa1AACTTTCAGCATGGCA</th>\n",
       "      <th>1dpa1AAGACCTGTTCGTCTC</th>\n",
       "      <th>...</th>\n",
       "      <th>samp2TTTCCTCGTCTAGTCA</th>\n",
       "      <th>samp2TTTCCTCGTTCATGGT</th>\n",
       "      <th>samp2TTTCCTCTCAGTTAGC</th>\n",
       "      <th>samp2TTTGCGCCACGCCAGT</th>\n",
       "      <th>samp2TTTGCGCTCACCGTAA</th>\n",
       "      <th>samp2TTTGGTTCATTCACTT</th>\n",
       "      <th>samp2TTTGTCACACCGGAAA</th>\n",
       "      <th>samp2TTTGTCACAGACAAAT</th>\n",
       "      <th>samp2TTTGTCAGTTATGTGC</th>\n",
       "      <th>samp2TTTGTCAGTTGTGGCC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSDARG00000000001</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSDARG00000000002</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSDARG00000000018</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSDARG00000000019</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSDARG00000000068</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24319</th>\n",
       "      <td>ENSDARG00000109183</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24320</th>\n",
       "      <td>ENSDARG00000109184</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24321</th>\n",
       "      <td>ENSDARG00000109185</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24322</th>\n",
       "      <td>ENSDARG00000109187</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24323</th>\n",
       "      <td>ENSDARG00000109188</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>24324 rows × 18552 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "samples               genes 1dpa1AAACCTGAGGGAAACA 1dpa1AAACCTGCACACTGCG  \\\n",
       "0        ENSDARG00000000001                     0                     0   \n",
       "1        ENSDARG00000000002                     0                     0   \n",
       "2        ENSDARG00000000018                     0                     0   \n",
       "3        ENSDARG00000000019                     0                     0   \n",
       "4        ENSDARG00000000068                     0                     0   \n",
       "...                     ...                   ...                   ...   \n",
       "24319    ENSDARG00000109183                   NaN                   NaN   \n",
       "24320    ENSDARG00000109184                     0                     0   \n",
       "24321    ENSDARG00000109185                     0                     0   \n",
       "24322    ENSDARG00000109187                     0                     0   \n",
       "24323    ENSDARG00000109188                     0                     0   \n",
       "\n",
       "samples 1dpa1AAACGGGGTATGAATG 1dpa1AAAGATGAGCGCCTCA 1dpa1AAAGTAGAGGACACCA  \\\n",
       "0                           0                     0                     0   \n",
       "1                           1                     0                     0   \n",
       "2                           0                     0                     0   \n",
       "3                           0                     0                     0   \n",
       "4                           0                     0                     0   \n",
       "...                       ...                   ...                   ...   \n",
       "24319                     NaN                   NaN                   NaN   \n",
       "24320                       0                     0                     0   \n",
       "24321                       0                     0                     0   \n",
       "24322                       0                     0                     0   \n",
       "24323                       0                     0                     0   \n",
       "\n",
       "samples 1dpa1AAAGTAGGTGTCGCTG 1dpa1AACGTTGGTGACGCCT 1dpa1AACTTTCAGCATGGCA  \\\n",
       "0                           0                     0                     0   \n",
       "1                           0                     0                     0   \n",
       "2                           0                     0                     0   \n",
       "3                           0                     0                     0   \n",
       "4                           0                     0                     0   \n",
       "...                       ...                   ...                   ...   \n",
       "24319                     NaN                   NaN                   NaN   \n",
       "24320                       0                     0                     0   \n",
       "24321                       0                     0                     0   \n",
       "24322                       0                     0                     0   \n",
       "24323                       0                     0                     0   \n",
       "\n",
       "samples 1dpa1AAGACCTGTTCGTCTC  ... samp2TTTCCTCGTCTAGTCA  \\\n",
       "0                           0  ...                     0   \n",
       "1                           0  ...                     0   \n",
       "2                           0  ...                     0   \n",
       "3                           0  ...                     0   \n",
       "4                           0  ...                     0   \n",
       "...                       ...  ...                   ...   \n",
       "24319                     NaN  ...                   NaN   \n",
       "24320                       0  ...                     0   \n",
       "24321                       0  ...                     0   \n",
       "24322                       0  ...                     0   \n",
       "24323                       0  ...                     0   \n",
       "\n",
       "samples samp2TTTCCTCGTTCATGGT samp2TTTCCTCTCAGTTAGC samp2TTTGCGCCACGCCAGT  \\\n",
       "0                           1                     0                     0   \n",
       "1                           0                     0                     0   \n",
       "2                           0                     0                     0   \n",
       "3                           0                     0                     0   \n",
       "4                           0                     0                     0   \n",
       "...                       ...                   ...                   ...   \n",
       "24319                     NaN                   NaN                   NaN   \n",
       "24320                       0                     0                     0   \n",
       "24321                       0                     0                     0   \n",
       "24322                       0                     0                     0   \n",
       "24323                       0                     0                     0   \n",
       "\n",
       "samples samp2TTTGCGCTCACCGTAA samp2TTTGGTTCATTCACTT samp2TTTGTCACACCGGAAA  \\\n",
       "0                           0                     0                     0   \n",
       "1                           0                     0                     0   \n",
       "2                           0                     0                     0   \n",
       "3                           0                     0                     0   \n",
       "4                           0                     0                     0   \n",
       "...                       ...                   ...                   ...   \n",
       "24319                     NaN                   NaN                   NaN   \n",
       "24320                       0                     0                     0   \n",
       "24321                       0                     0                     0   \n",
       "24322                       0                     0                     0   \n",
       "24323                       0                     0                     0   \n",
       "\n",
       "samples samp2TTTGTCACAGACAAAT samp2TTTGTCAGTTATGTGC samp2TTTGTCAGTTGTGGCC  \n",
       "0                           0                     0                     0  \n",
       "1                           0                     0                     0  \n",
       "2                           0                     0                     0  \n",
       "3                           1                     1                     0  \n",
       "4                           0                     0                     0  \n",
       "...                       ...                   ...                   ...  \n",
       "24319                     NaN                   NaN                   NaN  \n",
       "24320                       0                     0                     0  \n",
       "24321                       0                     0                     0  \n",
       "24322                       0                     0                     0  \n",
       "24323                       0                     0                     0  \n",
       "\n",
       "[24324 rows x 18552 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# transpose the final joined dataframe and set the header line and the index appropriately so that the input will work with rna sieve\n",
    "joined = joined.transpose()\n",
    "new_header = joined.iloc[0] #grab the first row for the header\n",
    "joined = joined[1:] #take the data less the header row\n",
    "joined.columns = new_header\n",
    "joined.reset_index(inplace=True)\n",
    "joined = joined.rename(columns = {'index':'genes'})\n",
    "joined"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# export the count file that now matches the design file\n",
    "joined_file = os.path.join('.', 'sorted_matched_Hou_shared_gene_counts.csv.gz')\n",
    "joined.to_csv(joined_file,\n",
    "           index=False,\n",
    "           compression=\"gzip\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# fill NaN values in both count files\n",
    "hartig_count = pd.read_csv(\"Hartig_shared_gene_counts.csv.gz\", compression=\"gzip\")\n",
    "hartig_count = hartig_count.fillna(0)\n",
    "hartig_file = os.path.join('.', 'fill_Hartig_shared_gene_counts.csv.gz')\n",
    "hartig_count.to_csv(hartig_file,\n",
    "           index=False,\n",
    "           compression=\"gzip\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "hou_count = pd.read_csv(\"sorted_matched_Hou_shared_gene_counts.csv.gz\", compression=\"gzip\")\n",
    "hou_count = hou_count.fillna(0)\n",
    "hou_file = os.path.join('.', 'fill_sorted_matched_Hou_shared_gene_counts.csv.gz')\n",
    "hou_count.to_csv(hou_file,\n",
    "           index=False,\n",
    "           compression=\"gzip\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}